tests/rest/client/test_retention.py

```diff
@@ -283,12 +290,15 @@ def test_state_policy(self):
 
         self._test_retention(room_id, expected_code_for_first_event=404)
 
-    def _test_retention(self, room_id, expected_code_for_first_event=200):
+    def _test_retention(
+        self, room_id: str, expected_code_for_first_event: int = 200
+    ) -> None:
         # Send a first event to the room. This is the event we'll want to be purged at the
         # end of the test.
         resp = self.helper.send(room_id=room_id, body="1", tok=self.token)
 
         first_event_id = resp.get("event_id")
+        assert first_event_id
```

Comment:
I think we normally prefer `assert first_event_id is not None` when we're narrowing `Optional[T]` to `T`.